src/Aspire.Hosting/Publishing/ContainerRuntimeBase.cs

```diff
@@ -64,17 +65,45 @@ await ExecuteContainerCommandAsync(
             imageName).ConfigureAwait(false);
     }
 
-    public virtual async Task PushImageAsync(string imageName, CancellationToken cancellationToken)
+    public virtual async Task PushImageAsync(IResource resource, CancellationToken cancellationToken)
     {
-        var arguments = $"push \"{imageName}\"";
-        
+        var pushOptions = await resource.ProcessImagePushOptionsCallbackAsync(
+            cancellationToken).ConfigureAwait(false);
+
+        // Try to get the container registry from DeploymentTargetAnnotation first
```

Comment:
Can/should this reuse ContainerImageReference? That way it can share the code below.